backend/src/post/Model.ts

```diff
@@ -0,0 +1,26 @@
+import { PrismaClient } from "@prisma/client";
+
+export class PostModel {
+  private db: PrismaClient;
+
+  constructor(db: PrismaClient) {
+    this.db = db;
+  }
+
+  async findAllPostsWithVotes() {
+    return await this.db.post.findMany({
+      include: {
+        votes: true, // Include associated votes for each post
```

Comment:
The code is clear to show other developers what happening, so no need to comment here